syntax/filter.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+/**
+ * DokuWiki Plugin struct (Syntax Component)
+ *
+ * @license GPL 2 http://www.gnu.org/licenses/gpl-2.0.html
+ * @author  Andreas Gohr, Michael Große <dokuwiki@cosmocode.de>
+ */
+
+use dokuwiki\plugin\struct\meta\AggregationCloud;
+use dokuwiki\plugin\struct\meta\ConfigParser;
+use dokuwiki\plugin\struct\meta\Filter;
+use dokuwiki\plugin\struct\meta\SearchConfig;
+use dokuwiki\plugin\struct\meta\StructException;
+
+class syntax_plugin_struct_filter extends DokuWiki_Syntax_Plugin
```

Comment:
when rebasing on #644 this probably should inherit from `syntax_plugin_struct_table`  and then `$this->illegalOptions` should be configured accordingly.